test/rest_api/test_storage_service.py

```diff
@@ -229,3 +229,100 @@ def test_storage_service_flush(cql, this_dc, rest_api):
 
                 resp = rest_api.send("POST", f"storage_service/keyspace_flush/{ks}", { "cf": f"{t0},{t1}"})
                 resp.raise_for_status()
+
+def test_storage_service_snapshot(cql, this_dc, rest_api):
+    resp = rest_api.send("GET", "storage_service/snapshots")
+    resp.raise_for_status()
+
+    tag_id = 0
+    def make_tag():
+        nonlocal tag_id
+        tag = f"test-snapshot-{tag_id}"
+        tag_id += 1
+        return tag
+
+    def verify_snapshot_details(expected):
+        resp = rest_api.send("GET", "storage_service/snapshots")
+        found = False
+        for data in resp.json():
+            if data['key'] == expected['key']:
+                assert not found
+                found = True
+                sort_key = lambda v: f"{v['ks']}-{v['cf']}"
+                value = sorted([v for v in data['value'] if not v['ks'].startswith('system')], key=sort_key)
+                expected_value = sorted(expected['value'], key=sort_key)
+                assert len(value) == len(expected_value), f"length mismatch: expected {expected_value} but got {value}"
+                for i in range(len(value)):
+                    v = value[i]
+                    ev = expected_value[i]
+                    assert v['ks'] == ev['ks'], f"ks mismatch: expected {expected_value} but got {value}"
+                    assert v['cf'] == ev['cf'], f"cf mismatch: expected {expected_value} but got {value}"
```

Comment:
The extra verbosity was helpful during debugging so I decided to leave it.
Without it, pytest would just report the different values at the specific assert but the context is unclear.
I prints the `expected` input, but in a concise form (with "..." if the printed output is too long)
and it doesn't print the `value` dictionary at all, so from the assert it is hard to tell what went wrong and why.

Reply:
Ah, and I can't simply compare the dicts to each other since the expected values for `total` and `live` are qualitative, not quantitative.  I don't know (and care about) what the actual size of the sstables is.
I just want to check whether `total` and `live` are either zero or non-zero in both `value` and `expected`.
Maybe if I normalize the values of `total` and `live` in the retrieved `value`, I could do a trivial dict comparison.

Reply:
> The extra verbosity was helpful during debugging so I decided to leave it. Without it, pytest would just report the different values at the specific assert but the context is unclear. I prints the `expected` input, but in a concise form (with "..." if the printed output is too long) and it doesn't print the `value` dictionary at all, so from the assert it is hard to tell what went wrong and why.

In my experience, it always prints the arrays being compared and the specific elements that were found different, even if sometimes you need to think a bit about what these elements mean.

Anyway, I won't insist here - I'll "resolve" this discussion. This was more of a general "warning" to people who switch to pytest from other systems like our old nosetest-based dtest or boost tests in C++, and got used to write all these elaborate comparison functions, while most of them aren't necessary in pytest (as you said, some normalization functions do come in handy, though, but those are usually much less verbose than comparison function, and you usually don't end up with a gazillion of them).